app.py

```diff
@@ -192,28 +192,72 @@ def reset_node_render_path(self, node):
+        the current script path and configuration.
         
         Note, this should really never be needed now that the
         path is reset automatically when the user changes something.
         """
         self.__write_node_handler.reset_render_path(node)
 
-    def convert_to_write_nodes(self):
+    def convert_to_write_nodes(self, show_warning=False):
         """
         Convert all Shotgun write nodes found in the current Script to regular
         Nuke Write nodes.  Additional toolkit information will be stored on 
         additional user knobs named 'tk_*'
+
+        :param show_warning: Optional bool that sets whether a warning box should be displayed to the user;
+         defaults to False.
         """
-        self.__write_node_handler.convert_sg_to_nuke_write_nodes()
 
-    def convert_from_write_nodes(self):
+        # By default we want to convert the write nodes, unless the warning is shown and the user chooses to abort.
+        continue_with_convert = True
+
+        if show_warning:
+            from sgtk.platform.qt import QtGui
+            res = QtGui.QMessageBox.question(None,
+                                             "Convert All SG Write Nodes?",
+                                             "Do you want to convert all SG write nodes to standard Nuke "
+                                             "write nodes?",
+                                             QtGui.QMessageBox.Yes | QtGui.QMessageBox.No)
+
+            if res != QtGui.QMessageBox.Yes:
+                # User chose to abort the operation, we should not convert the write nodes
+                continue_with_convert = False
+
+        if continue_with_convert:
+            self.__write_node_handler.convert_sg_to_nuke_write_nodes()
+
+    def convert_from_write_nodes(self, show_warning=False):
         """
         Convert all regular Nuke Write nodes that have previously been converted
         from Shotgun Write nodes, back into Shotgun Write nodes.
+
+        :param show_warning: Optional bool that sets whether a warning box should be displayed to the user;
+         defaults to False.
         """
-        self.__write_node_handler.convert_nuke_to_sg_write_nodes()
-    
+
+        # By default we want to convert the write nodes, unless the warning is shown and the user chooses to abort.
+        continue_with_convert = True
+
+        if show_warning:
+            from sgtk.platform.qt import QtGui
+            res = QtGui.QMessageBox.question(None,
+                                             "Convert All Write Nodes?",
+                                             "Do you want to convert all standard Nuke write nodes to "
+                                             "SG write nodes?"
+                                             "\n\nNote: It will only convert nodes that were previously SG write "
+                                             "nodes. It should also be considered experimental, some settings may not "
```

Comment:
as we discussed, we shouldn't show the convert actions in case the configuration contains any [promoted knobs](https://github.com/shotgunsoftware/tk-nuke-writenode/blob/master/info.yml#L76). 

Instad of saying "it's experimental" we should just not show the menu options if promoted knobs exist.